File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.*;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        long sum1 = 0;
        long sum2 = 0;

        Deque<Integer> q1 = new ArrayDeque<>();
        Deque<Integer> q2 = new ArrayDeque<>();
        for (int i = 0; i < queue1.length; ++i) {
            sum1 += queue1[i];
            q1.offer(queue1[i]);
            sum2 += queue2[i];
            q2.offer(queue2[i]);
        }

        int answer = 0;

        while (sum1 != sum2) {
            if (sum1 > sum2) {
                int q1Num = q1.poll();
                sum1 -= q1Num;
                sum2 += q1Num;
                q2.add(q1Num);
            } else if (sum1 < sum2) {
                int q2Num = q2.poll();
                sum2 -= q2Num;
                sum1 += q2Num;
                q1.add(q2Num);
            }
            
            if (answer >= queue1.length * 3)
                return -1;
            answer++;
        }

        return answer;
    }
}

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.cpp
#include <string>
#include <vector>
#include <queue>
#include <numeric>
using namespace std;

int solution(vector<int> queue1, vector<int> queue2) {
    int answer = 0;
   
    queue<int> q1, q2;
    
    long long sum1 = 0;
    long long sum2 = 0;
    
    for(int i = 0; i < queue1.size(); ++i)
    {
        q1.push(queue1[i]);
        sum1 += (long long)queue1[i];
        q2.push(queue2[i]);
        sum2 += (long long)queue2[i];
    }
    
    while(sum1 != sum2)
    {
        if (sum1 > sum2)
        {
            sum2 += (long long)q1.front();
            q2.push(q1.front());
            sum1 -= (long long)q1.front();
            q1.pop();
                    
        }
        else
        {
            sum1 += q2.front();
            q1.push(q2.front());
            sum2 -= q2.front();
            q2.pop();
        }
                    
        ++answer;
        
        if (answer > queue1.size() * 5)
            return -1;
    }
    
    
    return answer;
}